mitosheet/src/components/taskpanes/Concat/ConcatTaskpane.tsx

```diff
@@ -44,46 +45,63 @@ const ConcatTaskpane = (props: ConcatTaskpaneProps): JSX.Element => {
     // Make sure the user cannot select the newly created dataframe
     const [selectableSheetIndexes] = useState(props.sheetDataArray.map((sd, index) => index));
 
+    // For each sheet concatonated together, find all of the columns that are not included in the final result
+    const notIncludedColumnsArray = params?.sheet_indexes.map(sheetIndex => {
+        return Object.values(props.sheetDataArray[sheetIndex].columnIDsMap).filter(columnHeader => {
```

Comment:
`props.sheetDataArray[sheetIndex].columnIDsMap` to `props.sheetDataArray[sheetIndex]?.columnIDsMap || {}` for safety.

Reply:
Want to make this Object.values call just once and save it in a variable? Right now, it's O(n^3), methinks in the number of columns, at least - given all the copying.